src/huggingface_hub/_inference_endpoints.py

```diff
@@ -245,6 +245,7 @@ def update(
         revision: Optional[str] = None,
         task: Optional[str] = None,
         custom_image: Optional[Dict] = None,
+        secrets: Optional[Dict] = None,
```

Comment:
```suggestion
        secrets: Optional[Dict[str, str]] = None,
```

Let's make it obvious we want strings